hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/OzoneListStatusHelper.java

```diff
@@ -314,9 +314,8 @@ public OzoneFileStatus getStatus(String prefixPath, long scmBlockSize,
       } else {
         Preconditions.checkArgument(value instanceof OmKeyInfo);
         keyInfo = (OmKeyInfo) value;
-        keyInfo.setFileName(keyInfo.getKeyName());
         String fullKeyPath = OMFileRequest.getAbsolutePath(prefixPath,
-            keyInfo.getKeyName());
+            keyInfo.getFileName());
```

Comment:
Instead of doing this, I think it's better to have tool which should correct metadata. This will just avoid one problem but metadata will remain in corrupt state.

Reply:
You can refer tools like https://github.com/apache/ozone/pull/4337.